lib/matplotlib/legend_handler.py

```diff
@@ -749,6 +749,9 @@ def get_first(prop_array):
                 return None
         edgecolor = getattr(orig_handle, '_original_edgecolor',
                             orig_handle.get_edgecolor())
+        if edgecolor == 'face':
```

Comment:
Note we can't just get rid of the line before as it causes some hatching tests to fail for some reason...